doc/tunnel/SAI-Proposal-IP-in-MPLS-Tunnel.md

```diff
@@ -0,0 +1,776 @@
+# SAI IP-in-MPLS tunnel proposal
+
+
+-------------------------------------------------------------------------------
+ Title       | SAI IP-in-MPLS tunnel proposal
+-------------|-----------------------------------------------------------------
+ Authors     | Parvez Shaikh, Facebook Inc.
+ Status      | Under Review
+ Type        | Standards track
+ Created     | 03/23/2020
+ Updated     |
+ SAI-Version |
+
+-------------------------------------------------------------------------------
+
+
+## IP-in-MPLS network
+
+To implement either L3-VPN over MPLS or Segment Routing over MPLS (SR-MPLS), IP packets must be injected in MPLS domain at ingress. This involves the process of encapsulating IP packets into MPLS packets. At egress, traffic is decapsulated and resulting IP (or MPLS) packets are either forwarded to next hop or routed based on destination IP address.
```

Comment:
If this PR is meant for L3VPN use case, it would be nice to see an example SAI configuration for egress LER. I'm particularly interested in the SAI attribute which specifies the overlay(inner) VRF after the VPN label is popped-off at the last-hop.

Reply:
its already there, look for "vr0" in example, which depict object ID for virtual router. This is using existing MPLS router interface attribute, which indicates which VRF to use when top label is popped and then IP header is exposed, which virtual router to use for next L3 routing?